### DISCUSSION TITLE
Is it possible to write an plugin to extend the ability of the highlighted text?

### DISCUSSION THREAD

**Discussion by drewbaumann** [koreader/koreader#10240] [Q&A] [answered]
I am interested in developing plugins with similar behavior to how the wikipedia implementation works and noticed that feature modifies the following file directly: [frontend/apps/reader/modules/readerhighlight.lua](https://github.com/koreader/koreader/pull/818/files#diff-db51db4b48e0314c642050876de6cd601d148eaaa00d4511523f811531d63ccf)

I am curious if there is a way to add buttons that call back to functions purely through plugins without modifying core files.

Thank you!

**Comment by hius07** [comment]
Some discussion and ideas: https://github.com/koreader/koreader/pull/9416.

**Comment by poire-z** [comment, marked as answer]
Plugins should be able to use self.ui.highlight:addToHighlightDialog(...) to add buttons to the highlight dialog - although no plugin uses it currently.
#9746 added the same kind of thing so plugins can use `self.ui.link:addToExternalLinkDialog(...)` so their button is shown when long-press on a link - and plugins/wallabag.koplugin/main.lua does use it.

If this is not enough, you have the option of [User-patches ](https://github.com/koreader/koreader/wiki/User-patches) to possibly tweak core behaviour without modifying code code.

**Reply by drewbaumann** [reply]
Thank you. That gave me some helpful direction! It appears that this thread has a working example that I can use as inspiration: https://github.com/koreader/koreader/pull/6851
